src/main/java/jarvis/storage/TextDecoder.java

```diff
@@ -0,0 +1,49 @@
+package jarvis.storage;
+
+import jarvis.task.Deadline;
+import jarvis.task.Event;
+import jarvis.task.Task;
+import jarvis.task.Todo;
+
+/**
+ * Decodes the text file into a {@code Task} object.
+ */
+public class TextDecoder {
+
+    private static final int TASK_TYPE_INDEX = 1;
+    private static final int TASK_STATUS_INDEX = 4;
+    private static final int TASK_DESCRIPTION_START_INDEX = 7;
+
+    /**
+     * Decodes the text into a {@code Task} object.
+     *
+     * @param encodedTask task details in String format.
+     * @return {@code Task} object.
+     */
+    public static Task decodeTaskFromString(String encodedTask){
+        String taskDescription = encodedTask.substring(TASK_DESCRIPTION_START_INDEX);
+        Task task = new Task(taskDescription);
+        char taskType = encodedTask.charAt(TASK_TYPE_INDEX);
+        switch (taskType) {
+        case 'T':
```

Comment:
Perhaps you could try a more descriptive name for each `taskType`, to improve the code readability - single characters as names fall short of [naming conventions](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#not-too-long-not-too-short).